### ISSUE TITLE
coab strimzi - ingress failure

### ISSUE THREAD

**Issue by poblin-orange** [orange-cloudfoundry/paas-templates#1707]
### Expected behavior

* As a paas-templates operator | cf user | marketplace user | osb service provider | paas-templates author | paas-templates maintainer
* In order to 
* I need 

### Observed behavior

v53:
- v1.20.14+k3s1
- ingress-nginx-4.0.9
- strimzi-kafka-operator-0.28.0

1 single ingressclass is defined (nginx).

```
     I0901 14:04:09.365552       7 store.go:420] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-12e0d698-d87f-4601-80c7-14ab62e63683/kafka-cluster-kafka-0" error= │

    │ "ingress does not contain a valid IngressClass"                                                                                                                                                          │

    │ I0901 14:04:09.367649       7 admission.go:149] processed ingress via admission controller {testedIngressLength:1 testedIngressTime:0.116s renderingIngressLength:1 renderingIngressTime:0s admissionTim │

    │ e:18.4kBs testedConfigurationSize:0.116}                                                                                                                                                                 │

    │ I0901 14:04:09.367685       7 main.go:101] "successfully validated configuration, accepting" ingress="kafka-cluster-kafka-1/kafka-cluster-12e0d698-d87f-4601-80c7-14ab62e63683"                          │

    │ I0901 14:04:09.367960       7 store.go:420] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-12e0d698-d87f-4601-80c7-14ab62e63683/kafka-cluster-kafka-bootstrap │

    │ " error="ingress does not contain a valid IngressClass"                                                                                                                                                  │

    │ I0901 14:04:09.367994       7 store.go:420] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-12e0d698-d87f-4601-80c7-14ab62e63683/kafka-cluster-kafka-2" error= │

    │ "ingress does not contain a valid IngressClass"                                                                                                                                                          │

    │ I0901 14:04:09.376436       7 store.go:420] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-12e0d698-d87f-4601-80c7-14ab62e63683/kafka-cluster-kafka-1" error= │

    │ "ingress does not contain a valid IngressClass"                                                                                                                                                          │



```

### references
- https://github.com/strimzi/strimzi-kafka-operator/issues/4679
- https://github.com/strimzi/strimzi-kafka-operator/pull/1851
### workaround

explicitly set legacy nginx ingress annotation on Kafka CR listerners

 
### Affected releases

* x.y
* earlier versions

**Comment by poblin-orange** [comment]
Similar issue with v54.0.3 + k8s 1.23

```
│ I0901 18:48:51.506407       7 nginx.go:255] "Starting NGINX Ingress controller"                                                                                                                                 │
│ I0901 18:48:51.530740       7 event.go:282] Event(v1.ObjectReference{Kind:"ConfigMap", Namespace:"nginx-ingress", Name:"ingress-nginx-controller", UID:"b2aaaac7-8414-426f-a42a-e011b3265470", APIVersion:"v1", │
│ I0901 18:48:52.629136       7 store.go:423] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-kkkkkkk-aaaa-aaaa-aaaa-aaaaaaaaaaaa/kafka-cluster-kafka-bootstrap" error= │
│ I0901 18:48:52.629166       7 store.go:423] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-kkkkkkk-aaaa-aaaa-aaaa-aaaaaaaaaaaa/kafka-cluster-kafka-1" error="ingress │
│ I0901 18:48:52.629291       7 store.go:423] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-kkkkkkk-aaaa-aaaa-aaaa-aaaaaaaaaaaa/kafka-cluster-kafka-0" error="ingress │
│ I0901 18:48:52.629306       7 store.go:423] "Ignoring ingress because of error while validating ingress class" ingress="kafka-cluster-kkkkkkk-aaaa-aaaa-aaaa-aaaaaaaaaaaa/kafka-cluster-kafka-2" error="ingress │
│ I0901 18:48:52.710733       7 nginx.go:740] "Starting TLS proxy for SSL Passthrough"                                                                                                                            │

```

**Comment by poblin-orange** [comment]
fixed for 54.0.4
